buildSrc/build.gradle.kts

```diff
@@ -145,6 +145,8 @@ if (isSkipBuildSrcVerification) {
 }
 
 if (isCiServer) {
+    println("Current machine has ${Runtime.getRuntime().availableProcessors()} CPU cores")
+    require(Runtime.getRuntime().availableProcessors() >= 4) { "CI machines must have at least 4 CPU cores!" }
```

Comment:
Seems like we need to assert more CPU cores on Heztner boxes, any suggestions?

Reply:
Do we only have 4 cores on EC2?

How about we assert the 6 cores on Windows only?

Reply:
Do we get a build scan when we fail here?

Reply:
That bothers me a lot: we have the following archetectures:

- 8 cores on Hetzner Linux
- 8 cores on EC2 Linux
- 6 cores on Windows
- 4 cores on MacOS

> How about we assert the 6 cores on Windows only?

I'm not sure if that would be problem in the future on other platforms... ok, let's not do premature optimization.